OPHD/States/Planet.cpp

```diff
@@ -66,3 +81,80 @@ void Planet::update()
 	const auto spriteFrameOffset = NAS2D::Point{mTick % 8 * PlanetSize.x, ((mTick % 64) / 8) * PlanetSize.y};
 	renderer.drawSubImage(mImage, mPosition, NAS2D::Rectangle<int>::Create(spriteFrameOffset, PlanetSize));
 }
+
+
+void parseElementValue(Planet::PlanetType& destination, const NAS2D::Xml::XmlElement* element)
+{
+	destination = stringToEnum(Planet::planetTypeTable, element->getText());
+}
+
+
+void parseElementValue(Planet::Hostility& destination, const NAS2D::Xml::XmlElement* element)
+{
+	destination = stringToEnum(Planet::hostilityTable, element->getText());
+}
```

Comment:
This is probably a premature optimization. I want to move some large strings (definitions of the planets) into the xml file in the near future and I figured we may not want to pass large chunks of data in the return.

I'll make this a return for now as suggested and we can revisit if anyone ever notices a slowdown...

Reply:
EDIT: I was attempting to keep function names the same between parsing an enum and parsing a basic value. Since type deduction is not possible based on return values of a function, I think using a return value here isn't possible. The alternative if we want a return value would be to provide separate names for each function. I was hoping to keep function names the same to possibly allow collapsing if/else statement logic cases in future iterations.

Reply:
As of C++17, the compiler is guaranteed to do the optimization for you:
https://en.cppreference.com/w/cpp/language/copy_elision

When returning large objects, the compiler will effectively pass in a hidden parameter with the address of the variable being copied to, and will instead construct the value in place. Hence no copy during the return.

----

You can potentially use template methods to parameterize the return type. If the functions have different implementations, this can be handled with template specialization. Though often, just giving different functions different names may be easier and sometimes makes more sense. I'd need to look over the code again to consider this.